plugin/helpers_bots.go

```diff
@@ -72,3 +112,52 @@ func (p *HelpersImpl) EnsureBot(bot *model.Bot) (retBotId string, retErr error)
 
 	return createdBot.UserId, nil
 }
+
+func (p *HelpersImpl) ShouldProcessMessage(post *model.Post, botUserId string, options ...ShouldProcessMessageOption) (bool, error) {
+	messageProcessOptions := &shouldProcessMessageOptions{}
+	for _, option := range options {
+		option(messageProcessOptions)
+	}
+
+	if !messageProcessOptions.AllowSelf {
+		if post.UserId == botUserId {
+			return false, nil
+		}
+	}
+
+	if !messageProcessOptions.AllowSystemMessages {
+		if post.IsSystemMessage() {
+			return false, nil
+		}
+	}
+
+	if !utils.StringInSlice(post.ChannelId, messageProcessOptions.FilterChannelIDs) {
+		channel, appErr := p.API.GetChannel(post.ChannelId)
+
+		if appErr != nil {
+			p.API.LogError("Unable to get channel", "err", appErr)
```

Comment:
Please leave it to the caller to log the error. Otherwise we will end up with two log messages.

Reply:
We have done the same thing in `EnsureBot`. i.e here https://github.com/mattermost/mattermost-server/blob/e5ba0a0a1808660d4885ef04888da437c2a5b6a0/plugin/helpers_bots.go#L55
Do you want me to fix it here as well?

Reply:
The different in `EnsureBot` is this error is not returned. The rule is: either log the error or return it. The later is preferred.

Reply:
Okay. Thanks.

Reply:
I have made the changes.